File: 20221102/LeastCommonMultiple/src/main/java/LeastCommonMultiple.java
public class LeastCommonMultiple {
    public int solution(int[] arr) {
        int answer = arr[0];

        for (int number : arr) {
            answer = calcLeastCommonMultiple(answer, number);
        }
        return answer;
    }

    public int calcLeastCommonMultiple(int a, int b) {
        int greatestCommonDivisor = calcGreatestCommonDivisor(a, b);
        return a * b / greatestCommonDivisor;
    }

    public int calcGreatestCommonDivisor(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

File: 2022/20221103/LeastCommonMultiple.js
export function GCD(a, b) {
  return a % b ? GCD(b, a % b) : b;
}

export function LCM(a, b) {
  return a * b / GCD(a, b);
}

export default function solution(arr) {
  let answer = arr[0];

  arr.forEach((number) => {
    answer = LCM(answer, number);
  });
  return answer;
}
